tests/test_math.hh

```diff
@@ -59,6 +63,28 @@ namespace rascal {
     Eigen::Matrix<double, 3, Eigen::Dynamic>  results_airy;
     bool vebose{false};
   };
+
+  struct SphericalHarmonicsRefFixture {
+    SphericalHarmonicsRefFixture() {
+      json ref_data;
+      std::ifstream ref_file(this->ref_filename);
+      ref_file >> ref_data;
+      unit_vectors = ref_data.at("unit_vectors").get<StdVector2D_t>();
+      harmonics = ref_data.at("harmonics").get<StdVector3D_t>();
+      alps = ref_data.at("alps").get<StdVector3D_t>();
+    }
+
+    ~SphericalHarmonicsRefFixture() = default;
+
+    std::string ref_filename = "reference_data/spherical_harmonics_test.json";
+
+    using StdVector2D_t = std::vector<std::vector<double>>;
+    using StdVector3D_t = std::vector<std::vector<std::vector<double>>>;
+    StdVector2D_t unit_vectors{};
```

Comment:
I find the naming of `StdVector2D_t` confusing, but ok. I would expect the dimension to be inside the number of double-entries a vector has.

What I find more puzzling is that the unit_vectors are 2D. I do not insist on it, but I think a different naming of the types would benefit clarity?

Reply:
Yes, the naming is easily confused with Eigen's default-defined `VectorXd` types (where the D stands for double, not dimension). It's two-dimensional because it's a collection of unit vector_s_ (hence one dimension for Cartesian axis, one for collecting the vectors)